File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/services/account/ManageAccount.java
package com.decrux.pocketr.api.services.account;

import com.decrux.pocketr.api.entities.db.auth.User;
import com.decrux.pocketr.api.entities.dtos.AccountDto;
import com.decrux.pocketr.api.entities.dtos.CreateAccountDto;
import com.decrux.pocketr.api.entities.dtos.UpdateAccountDto;
import java.util.List;
import java.util.UUID;

public interface ManageAccount {

    AccountDto createAccount(CreateAccountDto dto, User owner);

    List<AccountDto> listIndividualAccounts(User owner);

    List<AccountDto> listAccountsByMode(User user, String mode, UUID householdId);

    AccountDto updateAccount(UUID id, UpdateAccountDto dto, User owner);
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr_api/services/account/ManageAccount.kt
package com.decrux.pocketr_api.services.account

import com.decrux.pocketr_api.entities.db.auth.User
import com.decrux.pocketr_api.entities.dtos.AccountDto
import com.decrux.pocketr_api.entities.dtos.CreateAccountDto
import com.decrux.pocketr_api.entities.dtos.UpdateAccountDto
import java.util.UUID

interface ManageAccount {

    fun createAccount(dto: CreateAccountDto, owner: User): AccountDto

    fun listAccounts(owner: User, includeArchived: Boolean = false): List<AccountDto>

    fun updateAccount(id: UUID, dto: UpdateAccountDto, owner: User): AccountDto
}
